frontend/src/components/ProfileContent.tsx

```diff
@@ -101,13 +173,39 @@ export default function ProfileContent(props: ProfileContentProps) {
                     <div className={classes.introVideo}>Intro Video</div>
                 </Grid>
             </Grid>
-            <Grid container item alignItems="center" className={classes.fieldsContainer} xs={12}>
-                {profileFields.map(({ title, value }: ProfileField) => (
-                    <Grid key={title} item className={classes.field} sm={12} md={6}>
-                        <Typography>{title}: {value}</Typography>
+            { isEditMode ? null :
+                <Grid id={readOnlyContent} container item alignItems="center" className={classes.fieldsContainer} xs={12}>
+                    <Button variant="contained" color="secondary" className={classes.bottomRight} onClick={handleEditBtnClick}>Edit</Button>
+                    <Grid id={readOnlyContent} container item alignItems="center" xs={12}>
+                        {Object.keys(profileFields).map(function (key) {
+                            let field: ProfileField = profileFields[key];
+                            const { title, value } = field;
+                            return (
+                                <Grid key={`${key}-readonlyFieldGrid`} item className={classes.field} sm={12} md={6}>
+                                    <Typography id={`${key}-readonlyFieldText`}>{title}: {value}</Typography>
+                                </Grid>
+                            );
+                        })}
                     </Grid>
-                ))}
-            </Grid>
+                </Grid>
+            }
+            { isEditMode ?
```

Comment:
I feel like you could hide the first and show the second if isEditMode=true,
and show the first and hide the second otherwise.

This would get rid of the messy `: null` result here.

Reply:
I used to know of a way to hide something like this easily but I cant find it online anywhere. the closest thing I found to do this without separating them into components is `visibility: hidden`, but that still renders the jsx and leaves room for it but just makes it invisible which is not what we want.

I'll look into better options

Reply:
Does `{isEditMode && etc}` work?

Reply:
Or maybe just one &?

Reply:
ahh yes, that worked great